# US BANK CUSTOMER

In [43]:
import pandas as pd
customer = pd.read_csv("https://raw.githubusercontent.com/junphp/is362_project2/master/US-Bank-Customers.csv")
customer

Customer ID         Name     Surname  Gender  Age     Region  \
0       200001088        Blake      Peters    Male   43  Charlotte   
1       200001739       Warren      Gibson    Male   52  Charlotte   
2       400002550         Lily       Baker  Female   23    Raleigh   
3       100002347      Leonard         Lee    Male   50       Cary   
4       200002349    Christian   Mackenzie    Male   59  Charlotte   
5       200002426        Gavin      McLean    Male   48  Charlotte   
6       100003895        Isaac    Mitchell    Male   38       Cary   
7       100000250       Amelia      Mathis  Female   30       Cary   
8       200000263    Gabrielle    Marshall  Female   41  Charlotte   
9       100000848          Tim    Hamilton    Male   27       Cary   
10      100000968       Edward      Martin    Male   38       Cary   
11      400001781        Irene     Chapman  Female   23    Raleigh   
12      100002704         Phil    Rampling    Male   45       Cary   
13      100002849         Anna      Peters  Female   51       Cary   
14      200000766        Keith        Ross    Male   45  Charlotte   
15      300001204       Olivia     Langdon  Female   42     Durham   
16      100001542      Dorothy       White  Female   30       Cary   
17      100001921        Donna    Marshall  Female   31       Cary   
18      200002579        Isaac      Wright    Male   46  Charlotte   
19      300003413  Christopher      Wright    Male   44     Durham   
20      200001693        Lucas    Clarkson    Male   48  Charlotte   
21      100002474        Chloe      Powell  Female   22       Cary   
22      200002752       Sophie      Howard  Female   57  Charlotte   
23      100002947        Faith       Poole  Female   49       Cary   
24      100002963        Donna       Allan  Female   37       Cary   
25      100000217      Carolyn        Hart  Female   28       Cary   
26      300002150      Jasmine    Mitchell  Female   37     Durham   
27      100000877        Megan        Reid  Female   34       Cary   
28      300003006       Claire       Scott  Female   34     Durham   
29      200003099        Colin       Walsh    Male   40  Charlotte   
...           ...          ...         ...     ...  ...        ...   
3984    100003259        Emily   Henderson  Female   37       Cary   
3985    100000971        Peter       Young    Male   23       Cary   
3986    100003023     Jonathan      Oliver    Male   37       Cary   
3987    100003229      Natalie  Sutherland  Female   37       Cary   
3988    200003650       Thomas      Fraser    Male   51  Charlotte   
3989    300000340      Rebecca        Nash  Female   20     Durham   
3990    300001339    Alexandra     Skinner  Female   40     Durham   
3991    100003376        Simon      Turner    Male   25       Cary   
3992    100004006     Kimberly        Gray  Female   44       Cary   
3993    100000898     Samantha        Ross  Female   47       Cary   
3994    100003173      Leonard        Gill    Male   39       Cary   
3995    100000338         Jake        Dyer    Male   42       Cary   
3996    200000661        Lucas     Abraham    Male   51  Charlotte   
3997    400001380         Ryan    Springer    Male   27    Raleigh   
3998    200001648       Andrew     Wilkins    Male   41  Charlotte   
3999    400001682       Lauren       Jones  Female   31    Raleigh   
4000    100002620       Rachel       Blake  Female   31       Cary   
4001    300003319       Gordon      Slater    Male   22     Durham   
4002    100003389        Sonia      Hunter  Female   33       Cary   
4003    200000018     Samantha     Coleman  Female   42  Charlotte   
4004    100000281          Sam     Parsons    Male   37       Cary   
4005    100002386       Andrea      Hudson  Female   46       Cary   
4006    100002562     Nicholas      Powell    Male   48       Cary   
4007    400000433         Rose   Robertson  Female   27    Raleigh   
4008    200002253        Diana       Terry  Female   50  Charlotte   
4009    10000234

## 1.question

** % of female/male customers who consider themselves white or blue collar **

In [44]:
customergroup = customer[["Job Classification","Gender"]].groupby(["Job Classification","Gender"]).agg({"Job Classification":"count"})
customergroup

Job Classification
Job Classification Gender                    
                   Female                 584
                   Male                   426
Blue Collar        Female                 255
                   Male                   794
White Collar       Female                1010
                   Male                   945

In [39]:
cust_pcts = customergroup.groupby(level=0).apply(lambda x: 100 * x / float(x.sum()))
cust_pcts

Job Classification
Job Classification Gender                    
                   Female           57.821782
                   Male             42.178218
Blue Collar        Female           24.308866
                   Male             75.691134
White Collar       Female           51.662404
                   Male             48.337596

## 1.Answer

** Blue collar Female 24.30% **

** Blue collar male   75.69% **

** White collar Female 51.66% **

** White collar male   48.33% **

## 2.Question

** % of married/single folks with a good credit rating (650 and above) **

In [95]:
customergroup2 = customer[(customer["Credit Rating"]>650) & (customer["Marriage Status"].str.contains('Single')==True)| (customer["Marriage Status"].str.contains('Married')==True)&(customer["Marriage Status"].str.contains('Married or married but separated')==False)&(customer["Marriage Status"].str.contains('Married or in consensual union')==False)&(customer["Marriage Status"].str.contains('Married, in consensual unions or separated')==False)]
customergroup2

Customer ID         Name    Surname  Gender  Age     Region  \
0       200001088        Blake     Peters    Male   43  Charlotte   
7       100000250       Amelia     Mathis  Female   30       Cary   
8       200000263    Gabrielle   Marshall  Female   41  Charlotte   
9       100000848          Tim   Hamilton    Male   27       Cary   
23      100002947        Faith      Poole  Female   49       Cary   
27      100000877        Megan       Reid  Female   34       Cary   
32      100001381    Sebastian      Allan    Male   30       Cary   
34      100003745     Caroline       Nash  Female   26       Cary   
35      100003917       Audrey      Walsh  Female   29       Cary   
41      100001820       Edward   Clarkson    Male   27       Cary   
44      100002302         Neil     Wright    Male   28       Cary   
63      100002505         John   Springer    Male   46       Cary   
64      100000489        Kevin     Wright    Male   41       Cary   
70      200002788       Trevor     Fraser    Male   53  Charlotte   
72      100003894          Tim    Chapman    Male   38       Cary   
83      100000278        Irene    Vaughan  Female   30       Cary   
84      100000618       Andrew     Wright    Male   42       Cary   
91      100002139         Luke  Mackenzie    Male   38       Cary   
96      400000442      Dorothy     Duncan  Female   33    Raleigh   
101     100002878         Emma      Short  Female   36       Cary   
115     100003397        Bella    Ellison  Female   30       Cary   
122     100002419        Diane    Ellison  Female   33       Cary   
125     200002394         Anna     Oliver  Female   44  Charlotte   
126     100002607         Ruth      Brown  Female   25       Cary   
128     300003104        Lucas     Newman    Male   42     Durham   
129     200003540       Nathan      Ogden    Male   57  Charlotte   
131     100000332      Carolyn  Churchill  Female   34       Cary   
137     100003856       Samson      James    Male   37       Cary   
139     400000396        Fiona   Buckland  Female   32    Raleigh   
142     100003458       Claire       Bond  Female   45       Cary   
...           ...          ...        ...     ...  ...        ...   
3900    200002147        Brian     Morgan    Male   63  Charlotte   
3903    100003438       Angela   Johnston  Female   32       Cary   
3905    200000792       Victor     Mathis    Male   55  Charlotte   
3915    200002399        Wendy   Ferguson  Female   57  Charlotte   
3922    100003272        Frank   Rampling    Male   41       Cary   
3923    200000944         Paul      Lyman    Male   47  Charlotte   
3929    100001195  Christopher      Walsh    Male   32       Cary   
3934    100003640         Adam      Short    Male   35       Cary   
3935    200001078        Isaac     Arnold    Male   47  Charlotte   
3939    100002918          Joe    Russell    Male   31       Cary   
3941    200003546        Gavin     Duncan    Male   56  Charlotte   
3942    300003549     Felicity   Lawrence  Female   44     Durham   
3944    200003898      William      Vance    Male   58  Charlotte   
3948    100002389     Felicity     Oliver  Female   45       Cary   
3949    100003048       Thomas   Springer    Male   46       Cary   
3950    100003345        Colin       Dyer    Male   38       Cary   
3953    200000947    Katherine       Ball  Female   56  Charlotte   
3954    400000988       Hannah   Hardacre  Female   39    Raleigh   
3956    200001180       Joanne      Ogden  Female   51  Charlotte   
3963    300001856     Virginia    Lambert  Female   30     Durham   
3967    200001330       Joshua      Black    Male   19  Charlotte   
3983    100003017      Michael    Pullman    Male   30       Cary   
3986    100003023     Jonathan     Oliver    Male   37       Cary   
3989    300000340      Rebecca       Nash  Female   20     Durham   
3993    100000898     Samantha       Ross  Female   47       Cary   
3996    200000661        Lucas    Abraham    Male   51  Charlotte   
4001

In [96]:
cust_rate=customergroup2.groupby("Marriage Status").agg({"Marriage Status":"count"})
cust_rate

Marriage Status
Marriage Status                 
Married                      736
Single                       255

In [101]:
credit_pcts = customergroup2.groupby("Marriage Status").agg({"Marriage Status":"count"}).apply(lambda x: 100 * x / float(x.sum()))
credit_pcts

Marriage Status
Marriage Status                 
Married                74.268416
Single                 25.731584

## 2.Answer

** Married 74.26% **

** Single  25.73% **

## 3.Question

** Gender with the most savings **

In [109]:
most_saving = customer.groupby("Gender").agg({"Balance":"sum"})
most_saving

Balance
Gender             
Female  72983534.07
Male    86638989.30

## 3.Answer 

** Male saving amount is great than Female saving amount **

## 4.Question

** Gender that is most employed **

In [110]:
most_employed = customer[(customer["Employed"].str.contains('Yes')==True)].groupby("Gender").agg({"Employed":"count"})
most_employed

Employed
Gender          
Female       621
Male         717

## 4.Answer

** Male is more employed than Female **

## 5.question

** City with the most highest number of employed customers **

In [111]:
city = customer[(customer["Employed"].str.contains('Yes')==True)].groupby("Region").agg({"Employed":"count"})
city

Employed
Region             
Cary            726
Charlotte       372
Durham          169
Raleigh          71

## 5. Answer

** Cary is most highest number of employed customers **